dimod/reference/samplers/exact_solver.py

```diff
@@ -85,6 +86,10 @@ def sample(self, bqm):
             :obj:`~dimod.SampleSet`
 
         """
+        for kw in kwargs:
+            msg = "Ignoring unknown kwarg: {!r}".format(kw)
+            warnings.warn(msg, RuntimeWarning)
```

Comment:
Perhaps subclass `UserWarning` as something like `UnknownSamplerArgumentWarning`, or similar, and raise that? FWIW, we do that in dwave-system for problem/solution warnings (e.g. `ChainBreakWarning` or `EnergyScaleWarning`).

Reply:
I would maybe change to SamplerUnknownArgumentWarning so it's clear it's the argument, not the sampler, that is unknown

Reply:
I added a SamplerUnknownArgWarning to sampler.py, but I'm wondering if there is a better place to put it.

Reply:
`dimod.exceptions` seems like a good place for now.